mainnet-contracts/src/PufferVaultV3.sol

```diff
@@ -0,0 +1,237 @@
+     * @param weth Address of the WETH token contract.
+     * @param lidoWithdrawalQueue Address of the Lido withdrawal queue contract.
+     * @param stETHStrategy Address of the stETH strategy contract.
+     * @param eigenStrategyManager Address of the EigenLayer strategy manager contract.
+     * @param oracle Address of the PufferOracle contract.
+     * @param delegationManager Address of the delegation manager contract.
+     * @param bridgingConstructorParams Constructor parameters for bridging.
+     */
+    constructor(
+        IStETH stETH,
+        IWETH weth,
+        ILidoWithdrawalQueue lidoWithdrawalQueue,
+        IStrategy stETHStrategy,
+        IEigenLayer eigenStrategyManager,
+        IPufferOracle oracle,
+        IDelegationManager delegationManager,
+        BridgingConstructorParams memory bridgingConstructorParams
+    ) PufferVaultV2(stETH, weth, lidoWithdrawalQueue, stETHStrategy, eigenStrategyManager, oracle, delegationManager) {
+        XTOKEN = IERC20(bridgingConstructorParams.xToken);
+        _DESTINATION_DOMAIN = bridgingConstructorParams.destinationDomain;
+        LOCKBOX = IXERC20Lockbox(bridgingConstructorParams.lockBox);
+        L2_REWARD_MANAGER = bridgingConstructorParams.l2RewardManager;
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Returns the total assets held by the vault.
+     * @dev Returns the total assets held by the vault, including ETH held in the eigenpods as a result of receiving rewards.
+     * @return The total assets held by the vault.
+     */
+    function totalAssets() public view virtual override returns (uint256) {
+        return (super.totalAssets() + getTotalRewardMintAmount());
+    }
+
+    /**
+     * @notice Returns the total reward mint amount.
+     * @return The total reward mint amount.
+     */
+    function getTotalRewardMintAmount() internal view returns (uint256) {
+        VaultStorage storage $ = _getPufferVaultStorage();
+        return $.totalRewardMintAmount;
+    }
+
+    /**
+     * @notice Mints and bridges rewards according to the provided parameters.
+     * @param params The parameters for bridging rewards.
+     */
+    function mintAndBridgeRewards(MintAndBridgeParams calldata params) external payable restricted {
+        VaultStorage storage $ = _getPufferVaultStorage();
+
+        if (params.rewardsAmount > $.allowedRewardMintAmount) {
+            revert InvalidMintAmount();
+        }
+
+        if (($.lastRewardMintTimestamp + $.allowedRewardMintFrequency) > block.timestamp) {
+            revert NotAllowedMintFrequency();
+        }
+
+        BridgeData memory bridgeData = $.bridges[params.bridge];
+
+        if (bridgeData.l2RewardManager == address(0)) {
+            revert BridgeNotAllowlisted();
+        }
+
+        uint256 ethToPufETHRate = previewDeposit(1 ether);
+        uint256 shares = ethToPufETHRate.mulDiv(params.rewardsAmount, 1 ether, Math.Rounding.Floor);
```

Comment:
why not `uint256 shares = convertToShares(params.rewardsAmount);`?

Reply:
Nice, can you pls make this comment in code, so that we don't get this question from the auditors